meta-lmp-base/classes/lmp.bbclass

```diff
@@ -47,53 +51,61 @@ IMAGE_CMD:ostree:append () {
+				"${cmd}" -repo "${OSTREE_REPO}" -creds "${SOTA_PACKED_CREDENTIALS}"
 			fi
-                else
-                        bbwarn "SOTA_PACKED_CREDENTIALS file does not exist."
-                fi
-        else
-                bbwarn "SOTA_PACKED_CREDENTIALS not set. Please add SOTA_PACKED_CREDENTIALS."
-        fi
+		else
+			bbwarn "SOTA_PACKED_CREDENTIALS file does not exist."
+		fi
+	else
+		bbwarn "SOTA_PACKED_CREDENTIALS not set. Please add SOTA_PACKED_CREDENTIALS."
+	fi
 }
 
 do_image_ostreepush[depends] += "ostreeuploader-native:do_populate_sysroot"
 IMAGE_CMD:ostreepush:prepend () {
 	if [ "${USE_FIOTOOLS}" = "1" ]; then
-		run_fiotool_cmd "fiopush"
-		# force return so garage-push called from meta-updater's IMAGE_CMD:ostreepush is not executed
+		# Fallback to the OE built fiopush if FIO_PUSH_CMD is not defined or it refers to non-existing or non-executable
+		run_fiotool_cmd "${FIO_PUSH_CMD}" "fiopush"
+		# force return so garage-push called from meta-updater's IMAGE_CMD_ostreepush is not executed
 		return
 	fi
 }
 
 do_image_garagecheck[depends] += "ostreeuploader-native:do_populate_sysroot"
 IMAGE_CMD:garagecheck:prepend () {
 	if [ "${USE_FIOTOOLS}" = "1" ]; then
-		run_fiotool_cmd "fiocheck"
-		# force return so garage-check called from meta-updater's IMAGE_CMD:garagecheck is not executed
+		# Fallback to the OE built fiocheck if FIO_CHECK_CMD is not defined or it refers to non-existing or non-executable
+		run_fiotool_cmd "${FIO_CHECK_CMD}" "fiocheck"
+		# force return so garage-check called from meta-updater's IMAGE_CMD_garagecheck is not executed
 		return
 	fi
 }
 
 ### Utilities
 
 def make_efi_dtb_boot_files(d):
-    # Generate IMAGE_EFI_BOOT_FILES entries for device tree files listed in
-    # KERNEL_DEVICETREE, to be available at the ESP/dtb folder (for u-boot).
-    # Use only the basename for dtb files:
-    alldtbs = d.getVar('KERNEL_DEVICETREE')
+	# Generate IMAGE_EFI_BOOT_FILES entries for device tree files listed in
+	# KERNEL_DEVICETREE, to be available at the ESP/dtb folder (for u-boot).
+	# Use only the basename for dtb files:
+	alldtbs = d.getVar('KERNEL_DEVICETREE')
 
-    # DTBs may be built out of kernel with devicetree.bbclass
-    if not alldtbs:
-        return ''
+	# DTBs may be built out of kernel with devicetree.bbclass
+	if not alldtbs:
+		return ''
 
-    def transform(dtb):
-        if not (dtb.endswith('dtb') or dtb.endswith('dtbo')):
-            bb.error("KERNEL_DEVICETREE entry %s is not a .dtb or .dtbo file." % (dtb) )
-        return os.path.basename(dtb) + ';dtb/' + os.path.basename(dtb)
+	def transform(dtb):
+		if not (dtb.endswith('dtb') or dtb.endswith('dtbo')):
+			bb.error("KERNEL_DEVICETREE entry %s is not a .dtb or .dtbo file." % (dtb) )
+		return os.path.basename(dtb) + ';dtb/' + os.path.basename(dtb)
 
-    return ' '.join([transform(dtb) for dtb in alldtbs.split() if dtb])
+	return ' '.join([transform(dtb) for dtb in alldtbs.split() if dtb])
```

Comment:
Can you drop these changes (in make_efi_dtb_boot_files) as part of this patch? Fine to have it done isolated in another patch, but we also need to make sure this won't cause problems as the logic is done in python.

Reply:
I've dropped them. Can make this fix in another commit or PR.